kubernetes/config/kube_config.c

```diff
@@ -393,8 +386,48 @@ int load_kube_config(char **pBasePath, sslConfig_t ** pSslConfig, list_t ** pApi
     }
 
   end:
+    return rc;
+}
+
+int load_kube_config(char **pBasePath, sslConfig_t ** pSslConfig, list_t ** pApiKeys, const char *configFileName)
+{
+    static char fname[] = "load_kube_config()";
+    int rc = 0;
+
+    kubeconfig_t *kubeconfig = kubeconfig_create();
```

Comment:
Given that you delete this below, I think you can stack allocate this and pass the pointer via `&` below.

That will save a heap allocation and also be cleaner code.

Reply:
This can be done but implies additional changes in kubeconfig_free() as this function fill attempt to free the kubeconfig_t* itself.  I can do a kubeconfig_free_static() if needed in order to keep the original api interface unchanged.

Please let me know your thoughts on this one.

Reply:
Meanwhile I added support to use both heap-based and stack-based allocated kubeconfig_t and a bit of documentation for these internal functions.